### DISCUSSION TITLE
Can I get MIDI In via serial DIN5 and have MIDI go via USB? [Raspberry Pi]

### DISCUSSION THREAD

**Discussion by fapplin** [FortySevenEffects/arduino_midi_library#292] [Q&A] [answered]
Hi,

With this library, can I get MIDI In via serial DIN5 and have MIDI go via USB? I'm kind of making a MIDI to USB adapter like you can buy on amazon, but need to process/massage some MIDI info before passing it out the USB.

BTW - I'm using a Raspberry Pi Pico and an Adafruit MIDI featherwing.

Thanks for any help.

Frank

**Comment by franky47** [comment, marked as answer]
With an Arduino board, you'd be able to use this library to do just that yes, unfortunately it's designed only for Arduino and not Raspberry Pi.

That being said, the transport mechanism is kind of platform-agnostic, so you theoretically could implement one for your serial interface and one for the USB driver and use the library as the MIDI parser & glue between the two transports.

**Comment by fapplin** [comment]
OK. Thanks for the info.
Frank

======================================================================= 
The price of greatness is responsibility. - Winston Churchill 

    On Thursday, July 21, 2022 at 07:56:32 AM EDT, François Best ***@***.***> wrote:  
 
 


With an Arduino board, you'd be able to use this library to do just that yes, unfortunately it's designed only for Arduino and not Raspberry Pi.

That being said, the transport mechanism is kind of platform-agnostic, so you could implement one for your serial interface and one for the USB driver and use the library as the MIDI parser & glue between the two transports.
